### ISSUE TITLE
Abort on non-string error in a lane

### ISSUE THREAD

**Issue by mpeterv** [LuaLanes/lanes#129]
```lua
local lanes = require "lanes"
lanes.configure()
lanes.gen("*", error)()
```

This throws `nil` in a lane and causes an abort due to an [assertion](https://github.com/LuaLanes/lanes/blob/0454f240bd443c75cf2a63544a6552a56ed132ff/src/lanes.c#L1944) that seems to check that the error message is a string.

As a workaround I'll wrap lane function in xpcall and turn errors into return values, but it would be convenient if non-string error objects, tables in particular, were properly handled.

**Comment by benoit-germain** [comment]
I'll give it a look, thanks for the report.

**Comment by benoit-germain** [comment]
Should be fixed by 16a76921157fbb7b8e4363bb56744ea2f3ac1156.

**Comment by mpeterv** [comment]
Indeed, thanks!
